rotkehlchen/api/rest.py

```diff
@@ -5009,3 +5010,47 @@ def get_event_details(self, identifier: int) -> Response:
             return api_response(wrap_in_fail_result('No details found'), status_code=HTTPStatus.NOT_FOUND)  # noqa: E501
 
         return api_response(_wrap_in_ok_result(details), status_code=HTTPStatus.OK)
+
+    def add_evm_transaction_hash(
+            self,
+            async_query: bool,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> Response:
+        if async_query is True:
+            return self._query_async(
+                command=self._add_evm_transaction_hash,
+                evm_chain=evm_chain,
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+
+        response = self._add_evm_transaction_hash(
+            evm_chain=evm_chain,
+            tx_hash=tx_hash,
+            associated_address=associated_address,
+        )
+        status_code = _get_status_code_from_async_response(response)
+        result_dict = {'result': response['result'], 'message': response['message']}
+        return api_response(process_result(result_dict), status_code=status_code)
+
+    def _add_evm_transaction_hash(
+            self,
+            evm_chain: SUPPORTED_CHAIN_IDS,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> dict[str, Any]:
+        evm_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain)
+        try:
+            transaction, tx_receipt = evm_manager.transactions.add_transaction_hash(
+                tx_hash=tx_hash,
+                associated_address=associated_address,
+            )
+        except (KeyError, DeserializationError, RemoteError, sqlcipher.IntegrityError) as e:  # pylint: disable=no-member  # noqa: E501
+            return wrap_in_fail_result(
+                message=f'unable to add transaction hash & address due to {str(e)}',
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
```

Comment:
Internal server error I believe is not the right code.

- For integrity error it should be: `HTTPStatus.CONFLICT`
- RemoteError, KeyError and DeserializationError should have `BAD_GATEWAY`

Adjust the docs properly

Reply:
Okay, for this... the integrity errors are caught at the serializer validation step.
- the check for in the transaction already exists and
- if the address is tracked.

and `BAD_REQUEST` is returned at that. So for consistency, I'd return bad request for `sqlcipher.IntegrityError` & bad gateway for remote error, keyError & deserializationError.